### PR TITLE
Use execution versioning for Move tools

### PR DESCRIPTION
**Pull request by dariorussi**
## Description 
Removing explicit usages of move dependencies in favor of `sui-execution`. 
That way we can version code and tests and use whichever version of Move we care about.
This specific PR is about `sui-move-build`.
I was piling PRs in #15030 and then realized those can go on their own and they will be even more clear. Moreover those are independent of creating a current (deployment) and latest (working) branches.
So there...
I need help to check that what I am doing is the right way to approach things.
I have added a `version` field to `BuildConfig` which defaults to `None` indicating the current production version.
If defined the given version will be used for everything versioned via `sui-execution`.
It would be great to hear if what is here makes sense...
@amnn @tnowacki @tzakian @stefan-mysten 

## Test Plan 

Curent tests

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-transaction-checks/src/lib.rs [hunk 1]
```diff
@@ -582,7 +582,9 @@ mod checked {
         // We use a custom config with metering enabled
         let is_metered = true;
         // Use the same verifier and meter for all packages
-        let mut verifier = sui_execution::verifier(protocol_config, is_metered, metrics);
+        let config = sui_execution::verifier_config(protocol_config, is_metered);
+        let execution_version = protocol_config.execution_version_as_option().unwrap_or(0);
```
Comment (dariorussi) [inline crates/sui-transaction-checks/src/lib.rs:586]: does this make sense?

File: crates/sui/src/client_commands.rs [hunk 2]
```diff
@@ -869,17 +871,23 @@ impl SuiClientCommands {
...
+                    build_config,
+                    package_path,
+                    protocol_config.execution_version_as_option(),
+                )?;
                 let modules: Vec<_> = package.get_modules().cloned().collect();
 
-                let mut verifier =
-                    sui_execution::verifier(&protocol_config, true, &bytecode_verifier_metrics);
+                let config = sui_execution::verifier_config(&protocol_config, true);
+                let execution_version = protocol_config.execution_version_as_option().unwrap_or(0);
```
Comment (dariorussi) [inline crates/sui/src/client_commands.rs:882]: does this make sense?

File: scripts/execution_layer.py [hunk 3]
```diff
@@ -467,6 +467,22 @@ def generate_lib(output_file: TextIO):
...
+    for (version, _, _) in cuts:
+        try:
+            ver = int(version)
+        except ValueError:
+            ver = -1
+        if latest_ver + 1 == ver:
+            latest_ver = ver
+        else:
+            break
+
```
Comment (dariorussi) [inline scripts/execution_layer.py:485]: @amnn and others, does this make sense?

File: scripts/execution_layer.py [hunk 4]
```diff
@@ -480,8 +496,12 @@ def substitute(m):
         elif var == "MOD_CUTS":
             return "".join(sorted(f"{spc}mod {cut};" for (_, _, cut) in cuts))
         elif var == "FEATURE_CONSTS":
-            return "".join(
-                f"{spc}pub const {feature}: u64 = {version};"
+            current_consts = (
+                f"{spc}pub const CURRENT: u64 = {latest_ver};\n"
+                f"pub const LATEST: u64 = {latest_ver};\n"
```
Comment (dariorussi) [inline scripts/execution_layer.py:501]: those look very useful to have, which is this whole shebang... thoughts?

**Review by tnowacki — COMMENTED** [review]
@dariorussi and I discussed this a good bit offline so writing up some soft/tentative conclusions we reached in our discussion: 

These changes are not going to be enough and it will get more and more sprawling (because of things like running unit tests), where we would really deeply need to thread through the versioning. Which is going to be immensely difficult. Luckily, it might make sense to never version the `sui move` commands and have them always use latest. The framework might have new natives or might rely on changes to the bytecode verifier, in which case it really can't be used against an old version. This is also true for user code (at least the verifier/language rule change bit)

Now what gets tricky: from a use perspective, you might install/re-install whatever development branch you are targeting (testnet v mainnet). If those are on "latest" for that branch, will that be valid with that branch? If not, we might have an issue there.

**Comment by dariorussi** [comment]
talk to @amnn and @tzakian as well and let's have a conversation at the next Move meeting about this.
Assuming `last` is the production (mainnet/testnet) version and `latest` the working version, we are considering the 2 options:
1- version all things that matter (definitely what is there today for execution + framework + compiler + ???) between `last` and `latest` and go through the effort/pain (not clear yet what the effort is but not minor at all) to be able to run both on CI, CLI, locally, etc.
2- leave things as they are, possibly clean up a bit more and make it easy to run an experiment as if the experiment was `latest` (current meaning of latest)

not clear what is better all things considered.
Let's think about it and talk next week

**Pull request closed without merging**